src/mesh_motion/MeshMotionAlg.C

```diff
@@ -61,8 +61,8 @@ void MeshMotionAlg::initialize( const double time )
 
     for (const auto& ff: fnames) {
       auto* fld = meta.get_field(stk::topology::NODE_RANK, ff);
-      fld->modify_on_host();
-      fld->sync_to_device();
+      fld->modify_on_device();
+      fld->sync_to_host();
     }
```

Comment:
Where was the field modified on device, and where is it needed on host?
In general you want the modify call as close as possible to where the actual modification happens, so that you only make the modify call if the field data was actually modified.
Similarly, put the sync_to_host call as close as possible to where the field is actually needed on host, so that you only do the sync if/when you actually need to. 
If these calls aren't placed carefully, it can cause un-necessary data copying between host and device, which is expensive.

Reply:
@alanw0 Thanks for pointing these out. I did plan on moving these over closer to where they should be but forgot to move them eventually. I have made the appropriate changes in 4a4789251b4a6edbf471d0671158a8cb3068bfdb